flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/BaseDeltaTaskWriter.java

```diff
@@ -57,6 +61,8 @@
     this.schema = schema;
     this.deleteSchema = TypeUtil.select(schema, Sets.newHashSet(equalityFieldIds));
     this.wrapper = new RowDataWrapper(flinkSchema, schema.asStruct());
+    this.keyWrapper =  new RowDataWrapper(FlinkSchemaUtil.convert(deleteSchema), deleteSchema.asStruct());
+    this.deleteSchemaProjection = RowDataProjection.create(schema, deleteSchema);
```

Comment:
Why do we just use the `structProjection` from the `BaseEqualityDeltaWriter` ?

Reply:
`structProjection` is part of `BaseEqualityDeltaWriter`, not `BaseTaskWriter`, which is what this class subclasses.

So it's not accessible, even when made protected.

Reply:
Resolving this comment. Please feel to reopen if need be. 🙂

Reply:
Will it be better to rename it `keyProjection` ?   Actually, here we don't try to generate a `deleteSchema` projection , instead we are trying to generate a `projection` wrapper to parse the key fields from the original `RowData`, right ?

Reply:
Yeah. The name `keyProjection` is a good call. More clear.

Reply:
Renamed.